submissions/ustymchyk/js-post-oop/index.js

```diff
@@ -118,64 +120,101 @@ class Human extends Creature {
 
     makeProposal(otherHuman) {
         if (!this.spouse && otherHuman.answerProposal(this)) {
-            this.spouse = otherHuman.name;
+            this.spouse = otherHuman;
         }
     }
 
     answerProposal(otherHuman) {
-        if (!this.spouse && this.#sympathy.includes(otherHuman.name)) {
-            this.spouse = otherHuman.name;
+        if (!this.spouse && this.#sympathy.includes(otherHuman)) {
+            this.spouse = otherHuman;
 
             return true;
         }
     }
 
     getFriendsNames() {
-        return this.#friends.join('; ');
+        return this.#friends.map(human => human.name).join('; ');
     }
 
-    introduce() {
+    toString() {
         const intro = `Hello! My name is ${this.name}, I'm ${this.age}. `;
-        let friends = '';
-        let spouse = '';
+        let friends = this.#friends.length ? `There is my friends: ${this.getFriendsNames()}. ` : '';
+        let spouse = this.spouse ? `And i have a spouse ${this.spouse.name}; ` : '';
+        let children = this.children.length ? `My ` + this.children.length > 1 ? `children: ${this._getChildren()}` : `child: ${this._getChildren()}` : '';
```

Comment:
Nested ternaries are easier to read when formatting reflects nesting levels.
Consider employing [prettier.io](https://prettier.io/) plugin to your IDE for it to take care of formatting and readability.
**This is not required to fix now.**